klippy/extras/tmc.py

```diff
@@ -173,20 +177,27 @@ def __init__(self, mcu_tmc, mcu_endstop):
         self.query_endstop_wait = self.mcu_endstop.query_endstop_wait
         self.TimeoutError = self.mcu_endstop.TimeoutError
     def home_prepare(self):
-        self.fields.set_field("en_pwm_mode", 0)
-        val = self.fields.set_field("diag1_stall", 1)
+        if self.tmc_type == 2130 or self.tmc_type == 5160:
+            val = self.fields.set_field("diag1_stall", 1)
+            val = self.fields.set_field("en_pwm_mode", 0)
+        elif self.tmc_type == 2209:
+            val = self.fields.set_field("en_spreadCycle", 1)
```

Comment:
This code does not look complete to me.  If I understand it correctly, the tmc2209 requires en_spreadCycle=1, TCOOLTHRS=0xfffff, and TPWMTHRS=0.  All must then be restored on home complete.

Also, FWIW, en_spreadCycle is the inverse of en_pwm_mode, so I think it's a little confusing to save en_spreadCycle in a field named self.en_pwm.

Reply:
TPWMTHRS is never set in klipper, so i skipped it as tmc2208 , tmc2130 implementation does. Of course i can add it. So a current state can be saved and restored but it will save 0, write 0 and restore 0. So i thought thats the reason its not there in 2130. Thats why i skipped it in 5160. so i did that for 2209 too. I can add it if i did not see the reason.

-

self.en_pwm is renamed now to self.pwm_state

Reply:
TPWMTHRS is set in TMCStealthchopHelper() - it's the stealthchop_threshold.  It's set on tmc2208, tmc2130, and tmc5160.

Reply:
Okay i add that but why not for TMC2130 too?

Reply:
I'm not sure what you are referring to with the tmc2130.

-Kevin